python-sdk/src/astro/utils/jinja.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from jinja2.environment import Environment
+from jinja2.loaders import FileSystemLoader
+from jinja2.meta import find_undeclared_variables
+from jinja2.runtime import StrictUndefined
+
+
+def find_template_variables(file_path: Path) -> set[str]:
+    """
+    Find template variables in given file path which needed to be declared when rendering.
+
+    :param file_path: The file path to check for variables.
+
+    :returns: all undeclared variables.
+    """
+    env = Environment(
+        loader=FileSystemLoader(file_path.parent),
```

Comment:
Does this work well when the templates are not kept in this parent directory but are in the sql-cli directory? We were facing some challenges today where in it was giving TemplateNotFound Error.

cc: @feluelle

Reply:
@feluelle can you explain this further?

Reply:
@pankajkoti this is `find_template_variables` - not the `render` function. It does not run `get_template`. This is completely fine like that.